plasmapy/simulation/particle_tracker.py

```diff
@@ -0,0 +1,883 @@
+                        f"non-physical effects, but a value of {edge_max:.2E} {unit} was "
+                        f"found on the edge of the {rq} array. Consider applying a "
+                        "envelope function to force the fields at the edge to go to "
+                        "zero.",
+                        RuntimeWarning,
+                    )
+
+    @property
+    def num_grids(self):
+        """The number of grids specified at instantiation."""
+        return len(self.grids)
+
+    def _log(self, msg):
+        if self.verbose:
+            print(msg)  # noqa: T201
+
+    @particle_input
+    def load_particles(
+        self,
+        x,
+        v,
+        particle: Particle = Particle("p+"),  # noqa: B008
+    ):
+        r"""
+        Load arrays of particle positions and velocities.
+
+        Parameters
+        ----------
+        x : `~astropy.units.Quantity`, shape (N,3)
+            Positions for N particles
+
+        v : `~astropy.units.Quantity`, shape (N,3)
+            Velocities for N particles
+
+        particle : |particle-like|, optional
+            Representation of the particle species as either a |Particle| object
+            or a string representation. The default particle is protons.
+
+        distribution : str
+            A keyword which determines how particles will be distributed
+            in velocity space. Options are:
+
+                - 'monte-carlo': velocities will be chosen randomly,
+                    such that the flux per solid angle is uniform.
+
+                - 'uniform': velocities will be distributed such that,
+                   left unperturbed,they will form a uniform pattern
+                   on the detection plane.
+
+            Simulations run in the ``'uniform'`` mode will imprint a grid pattern
+            on the image, but will well-sample the field grid with a
+            smaller number of particles. The default is ``'monte-carlo'``.
+
+
+        """
+        # Raise an error if the run method has already been called.
+        self._enforce_order()
+
+        self.q = particle.charge.to(u.C).value
+        self.m = particle.mass.to(u.kg).value
+
+        if x.shape[0] != v.shape[0]:
+            raise ValueError(
+                "Provided x and v arrays have inconsistent numbers "
+                " of particles "
+                f"({x.shape[0]} and {v.shape[0]} respectively)."
+            )
+        else:
+            self.nparticles = x.shape[0]
+
+        self.x = x.to(u.m).value
+        self.v = v.to(u.m / u.s).value
+
+    def _stop_particles(self, particles_to_stop_mask):
+        if len(particles_to_stop_mask) != self.x.shape[0]:
+            raise ValueError(
+                f"Expected mask of size {self.x.shape[0]}, got {len(particles_to_stop_mask)}"
+            )
+
+        self.v[particles_to_stop_mask] = np.NaN
+
+    def _remove_particles(self, particles_to_remove_mask):
```

Comment:
Write just one test each for _remove_particles and _stop_particles that shows they work as intended (to satisfy codecov)

Reply:
Should I keep these methods in this PR?? They aren't being used yet

Reply:
I think that's ok - since they're pretty basic functionality for a particle tracer (and we're just going to add them in the immediate next PR anyway)